pkg/kn/traffic/compute_test.go

```diff
@@ -51,169 +53,188 @@ func newTestTrafficCommand() (*cobra.Command, *flags.Traffic) {
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "", 100, true)),
 			[]string{"--tag", "@latest=current,echo-v0=stale,echo-v1=old"},
 			[]string{"@latest", "echo-v0", "echo-v1"},
 			[]string{"current", "stale", "old"},
 			[]int{100, 0, 0},
 		},
-		// re-assign same tag to echo-v1
 		{
+			"re-assign same tag to 'echo-v1' revision",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--tag", "echo-v1=latest"},
 			[]string{"echo-v1"},
 			[]string{"latest"},
 			[]int{100},
 		},
-		// set 2% to @latest (append it in traffic block)
 		{
+			"set 2% traffic to latest revision by appending it in traffic block",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--traffic", "@latest=2,echo-v1=98"},
 			[]string{"echo-v1", "@latest"},
 			[]string{"latest", ""},
 			[]int{98, 2},
 		},
-		// set 2% to @latest with tag (append it in traffic block)
 		{
+			"set 2% to @latest with tag (append it in traffic block)",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--traffic", "@latest=2,echo-v1=98", "--tag", "@latest=testing"},
 			[]string{"echo-v1", "@latest"},
 			[]string{"latest", "testing"},
 			[]int{98, 2},
 		},
-		// change traffic percent of an existing revision in traffic block, add new revision with traffic share
 		{
+			"change traffic percent of an existing revision in traffic block, add new revision with traffic share",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("v1", "echo-v1", 100, false)),
 			[]string{"--tag", "echo-v2=v2", "--traffic", "v1=10,v2=90"},
 			[]string{"echo-v1", "echo-v2"},
 			[]string{"v1", "v2"},
 			[]int{10, 90}, //default value,
 		},
-		// append new non-@latest target
-		{
-			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("v1", "echo-v1", 100, false)),
-			[]string{"--tag", "echo-v2=v2", "--traffic", "v1=10,v2=90"},
-			[]string{"echo-v1", "echo-v2"},
-			[]string{"v1", "v2"},
-			[]int{10, 90}, //default value,
-		},
-		// untag
 		{
+			"untag 'latest' tag from 'echo-v1' revision",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--untag", "latest"},
 			[]string{"echo-v1"},
 			[]string{""},
 			[]int{100},
 		},
+		{
+			"untag 'latest' tag from '@latst' revision",
+			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "", 100, true)),
+			[]string{"--untag", "latest"},
+			[]string{"@latest"},
+			[]string{""},
+			[]int{100},
+		},
+		{
+			"replace revision pointing to 'latest' tag from 'echo-v1' to 'echo-v2' revision",
+			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 50, false), newTarget("", "echo-v2", 50, false)),
+			[]string{"--untag", "latest", "--tag", "echo-v1=old,echo-v2=latest"},
```

Comment:
again, doesn't additionally test any additional combination, the individual operations are tested in earlier tests with different combinations.

Reply:
> the individual operations are tested in earlier tests with different combinations.

That's true, and my motivation to introduce this test case is that I would like to cover all the combinations described in Example section in the [docs](https://github.com/knative/client/pull/331).

Reply:
Fine.

I planned to cover the examples in docs in e2e and have unit tests cover to *unit* functionality. I am fine with keeping it.